mk/kafka.mk

```diff
@@ -137,7 +61,7 @@ kafka-produce-msg: KAFKA_MESSAGE_KEY ?= c67cd587-3741-493d-9302-f655fcd3bd68
 kafka-produce-msg: KAFKA_MESSAGE_FILE ?= test/kafka/demo-introspect-request-1.json
 kafka-produce-msg: ## Produce a demo kafka message to introspect
 	$(DOCKER) run \
-	  --net container:kafka-zookeeper \
+	  --net container:$(ZOOKEEPER_CONTAINER_NAME) \
```

Comment:
I was not able to run this command from a container because of the name resolution of the kafka broker and the list of hosts reported by kafka broker when negotiated the connection (I tried with different networks: default, bridge, host, cs_default and none of them worked).

This add a requirement to get installed `kcat`tool into the workstation.

I suggest:
- Add a note into README.md to install `kcat` tool (https://github.com/edenhill/kcat#install).
- Update this rule with the content below:

```makefile
# The tool cannot be run from a container because of name resolution
# situations, so it should be installed into the host machine and
# run it directly
# https://github.com/edenhill/kcat#install
# https://stackoverflow.com/questions/58716683/is-there-a-way-to-add-headers-in-kafka-console-producer-sh
# https://github.com/edenhill/kcat
# https://dev.to/de_maric/learn-how-to-use-kafkacat-the-most-versatile-kafka-cli-client-1kb4
.PHONY: kafka-produce-msg
kafka-produce-msg: KAFKA_TOPIC ?= $(firstword $(KAFKA_TOPICS))
kafka-produce-msg: KAFKA_IDENTITY ?= eyJpZGVudGl0eSI6eyJ0eXBlIjoiQXNzb2NpYXRlIiwiYWNjb3VudF9udW1iZXIiOiIxMTExMTEiLCJpbnRlcm5hbCI6eyJvcmdfaWQiOiIyMjIyMjIifX19Cg==
kafka-produce-msg: KAFKA_REQUEST_ID ?= demo
kafka-produce-msg: KAFKA_MESSAGE_KEY ?= c67cd587-3741-493d-9302-f655fcd3bd68
kafka-produce-msg: KAFKA_MESSAGE_FILE ?= test/kafka/demo-introspect-request-1.json
kafka-produce-msg: ## Produce a demo kafka message to introspect
	kcat \
	  -k "$(KAFKA_MESSAGE_KEY)" \
	  -H X-Rh-Identity="$(KAFKA_IDENTITY)" \
	  -H X-Rh-Insight-Request-Id="$(KAFKA_REQUEST_ID)" \
	  -H Type="Introspect" \
	  -b localhost:9092 \
	  -t $(KAFKA_TOPIC) \
	  -P <<< "$$(cat "$(KAFKA_MESSAGE_FILE)" | jq -c -M )"

```

Reply:
I got the container version working for me while using podman, curious if it works for you?

Reply:
sorry, I didn't see the last comment! it worked for linux + podman, it failed for macos + docker

Reply:
I correct myself! It worked with podman+linux and macos+docker! my last try I didn't override DOCKER=docker (I have docker and podman installed in macos).